packaging/vagrant/Vagrantfile

```diff
@@ -0,0 +1,229 @@
+# This Vagrantfile builds a Centos 7 virtual machine with Charliecloud and
+# Docker installed. It is documented in the HTML docs.
+
+# A few design decisions:
+#
+# 1. We use the CentOS kernel (3.10 + lots of Red Hat patches) because it's
+#    easier than installing the upstream kernel and seems to work fine once
+#    user namespaces are turned on. Please let us know of any problems.
+#
+# 2. /tmp is the CentOS default: simply a directory on the root filesystem,
+#    but cleaned out on each boot. This is to avoid (a) configuring it to be a
+#    tmpfs and (b) to save swap.
+#
+# 3. We don't try any clever sizing of the VM appliance (e.g., number of
+#    vCPUs, RAM). End users should adjust these values, but we want to leave
+#    reasonable defaults in case they don't, even if they have a smallish
+#    host. For example, we don't want to configure the appliance to take
+#    advantage of your 96-core Beast Machine and then give it to some poor end
+#    user to run that 96-vCPU VM on their 4-core laptop.
+#
+# 4. We set the VirtualBox GUI to "true", i.e., present a console window, for
+#    two reasons. First, this provides a method for users to get in and change
+#    their password without us needing to hand out a temporary password
+#    (auto-login on the first console). Second, it's friendlier for less
+#    technical end users. Power users can turn it off later if they prefer.
+#
+# 5. No OpenMPI is installed. This saves build time, a little disk space, and
+#    the complexity of finding the right version to match the Charliecloud
+#    examples, while the cost is low: a few tests are skipped. Users who want
+#    to run single-node MPI apps in the
+
+# Prerequisites.
+Vagrant.require_version ">= 2.1.0"
+["disksize", "proxyconf", "reload", "vbguest"].each do |p|
+  if not Vagrant.has_plugin?("vagrant-#{p}")
+    raise("plugin vagrant-#{p} is required")
+  end
+end
+
+Vagrant.configure("2") do |c|
+
+  c.vm.box = "centos/7"  # https://app.vagrantup.com/centos/boxes/7
+  c.vm.hostname = "charliebox"
+
+  # Note: Vagrant sets up a port mapping from localhost:2222 to charliebox:22
+  # automatically, so we need no repeat that here.
+
+  # Set up proxies if appropriate.
+  if ENV["HTTP_PROXY"] or ENV["HTTPS_PROXY"] or ENV["NO_PROXY"]
+    if not (ENV["HTTP_PROXY"] and ENV["HTTPS_PROXY"] and ENV["NO_PROXY"])
+      raise("missing proxy variable(s)")
+    end
+    c.proxy.http = ENV["HTTP_PROXY"]
+    c.proxy.https = ENV["HTTPS_PROXY"]
+    c.proxy.no_proxy = ENV["NO_PROXY"]
+    c.vm.provision "proxy", type:"shell", privileged: true, inline: <<-EOF
+      echo 'Defaults env_keep+="ALL_PROXY all_proxy auto_proxy RSYNC_PROXY"' \
+          >> /etc/sudoers.d/proxy
+    EOF
+  end
+
+  # FIXME: Why is this false? Vagrant recommends true.
+  c.vm.box_check_update = false
```

Comment:
The user can always destroy and re-provision the VM if they choose, but because the Charliecloud VM takes awhile to fully provision, we don't want to inflict unexpected re-provisions on the user.